File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    enum = {}
    answer = []
    for i in range(0, len(name), 1):
        enum[name[i]] = yearning[i]
    for i in photo:
        cnt = 0
        for j in i:
            if enum.get(j) is not None:
                cnt += enum.get(j)
        answer.append(cnt)

    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        HashMap<String, Integer> tempMap = new HashMap<String, Integer>();
        for (int i = 0; i < name.length; i++) {
            tempMap.put(name[i], yearning[i]);
        }
        int[] pointArray = new int[photo.length];
        for (int i = 0; i < photo.length; i++) {
            int tempPoint = 0;
            for (String human : photo[i]) {
                Integer temp = tempMap.get(human);
                if (temp != null) {
                    tempPoint += tempMap.get(human);
                }
            }
            pointArray[i] = tempPoint;
        }
        return pointArray;
    }
}